src/ui/common/src/components/workflows/artifact/content.tsx

```diff
@@ -56,57 +56,83 @@ const ArtifactContent: React.FC<Props> = ({
   }
 
   if (!contentWithLoadingStatus.data) {
-    return (
-      <Typography variant="h5" component="div" marginBottom="8px">
-        No result to show for this artifact.
-      </Typography>
-    );
-  }
-
-  if (
-    artifact.type === ArtifactType.Bytes ||
```

Comment:
I feel this makes the switch statement below slightly hard to read. I would consider doing the following instead:
* put Bytes and Picklable inside the switch below so that this switch cover all types. For these two types we don't have to check if the data field exists
* For those we expected to have data filled (e.g. images, table), if the data doesn't exists, we show an alert since this should be considered as an error case.

Reply:
The checking of whether data field is empty happens before the switch statement, so if I put these two cases inside the switch statement they will never be hit right?

Reply:
Yeah I mean this way we will completely remove the empty field check in line 58 and add those for each case in the switch. Since you already build error handling for image and table, it should be fine to just add that to json case